exotica/src/Problems/UnconstrainedEndPoseProblem.cpp

```diff
@@ -84,16 +84,44 @@ void UnconstrainedEndPoseProblem::Instantiate(UnconstrainedEndPoseProblemInitial
 
     if (init.Rho.rows() == NumTasks) Rho = init.Rho;
     if (init.Goal.rows() == PhiN) y.data = init.Goal;
+
+    S = Eigen::MatrixXd::Identity(JN, JN);
+    ydiff = Eigen::VectorXd::Zero(JN);
+}
+
+void UnconstrainedEndPoseProblem::preupdate()
+{
+    PlanningProblem::preupdate();
+    for (TaskMap_ptr task : Tasks)
+    {
+        for (int i = 0; i < task->Length; i++)
+        {
+            S(i + task->Start, i + task->Start) = Rho(task->Id);
+        }
+    }
+}
+
+double UnconstrainedEndPoseProblem::getScalarCost()
+{
+    return ydiff.transpose()*S*ydiff;
+}
+
+Eigen::VectorXd UnconstrainedEndPoseProblem::getScalarJacobian()
+{
+    return J.transpose()*S*ydiff*2.0;
 }
 
 void UnconstrainedEndPoseProblem::Update(Eigen::VectorXdRefConst x)
 {
     scene_->Update(x);
+    Phi.setZero(PhiN);
+    J.setZero();
```

Comment:
This is currently done in each individual taskmap. We should follow up and remove the unnecessary setting to zero there.